android/src/main/java/com/getcapacitor/community/admob/executors/AdRewardExecutor.java

```diff
@@ -29,83 +32,22 @@ public AdRewardExecutor(
-            }
-
-            @Override
-            public void onRewardedVideoAdOpened() {
-                notifyListenersFunction.accept(RewardedAdEventName.onRewardedVideoAdOpened.name(), new JSObject().put("value", true));
-            }
-
-            @Override
-            public void onRewardedVideoStarted() {
-                notifyListenersFunction.accept(RewardedAdEventName.onRewardedVideoStarted.name(), new JSObject().put("value", true));
-            }
-
-            @Override
-            public void onRewardedVideoAdClosed() {
-                notifyListenersFunction.accept(RewardedAdEventName.onRewardedVideoAdClosed.name(), new JSObject().put("value", true));
-            }
-
-            @Override
-            public void onRewarded(RewardItem rewardItem) {
-                final JSObject adMobRewardItem = new JSObject();
-                adMobRewardItem.put("type", rewardItem.getType());
-                adMobRewardItem.put("amount", rewardItem.getAmount());
-                notifyListenersFunction.accept(RewardedAdEventName.onRewarded.name(), adMobRewardItem);
-            }
-
-            @Override
-            public void onRewardedVideoAdLeftApplication() {
-                notifyListenersFunction.accept(
-                    RewardedAdEventName.onRewardedVideoAdLeftApplication.name(),
-                    new JSObject().put("value", true)
-                );
-            }
-
-            @Override
-            public void onRewardedVideoAdFailedToLoad(int errorCode) {
-                JSObject adMobError = new JSObject();
-                adMobError.put("code", errorCode);
-                adMobError.put("reason", RequestHelper.getRequestErrorReason(errorCode));
-                notifyListenersFunction.accept(RewardedAdEventName.onRewardedVideoAdFailedToLoad.name(), adMobError);
-            }
-
-            @Override
-            public void onRewardedVideoCompleted() {
-                notifyListenersFunction.accept(RewardedAdEventName.onRewardedVideoCompleted.name(), new JSObject().put("value", true));
-            }
-        };
-    }
-
+    @PluginMethod
     public void prepareRewardVideoAd(final PluginCall call, BiConsumer<String, JSObject> notifyListenersFunction) {
         final AdOptions adOptions = AdOptions.getFactory().createRewardVideoOptions(call);
 
         try {
-            mRewardedVideoAd = MobileAds.getRewardedVideoAdInstance(contextSupplier.get());
-
             activitySupplier
                 .get()
                 .runOnUiThread(
                     () -> {
                         final AdRequest adRequest = RequestHelper.createRequest(adOptions);
                         final String id = AdViewIdHelper.getFinalAdId(adOptions, adRequest, logTag, contextSupplier.get());
-                        mRewardedVideoAd.loadAd(id, adRequest);
-                        mRewardedVideoAd.setRewardedVideoAdListener(getRewardedVideoAdListener(call, notifyListenersFunction));
```

Comment:
@rdlabo I am having troubles understanding the API change for `RewardAds`. The notifications for successfull Reward item is removed from `definitions.ts` here and the implementation is also removed here from the Executor. 

https://github.com/capacitor-community/admob/commit/93182951c96dee8df4cc288ea80466391e49192b#diff-f1f3c6677661c21128a30f64358c6615c6801043e1cbfcc904302422a17b3246L112-L115

That is the reason why the AdRewardExecutor test is failing here:

![image](https://user-images.githubusercontent.com/5490201/114300302-ebc63a80-9abf-11eb-861c-ae2f7dd3a26c.png)

Why are those events removed?

Reply:
@distante New Reward API can get item from `showRewardVideoAd(item)`.
https://github.com/capacitor-community/admob/commit/93182951c96dee8df4cc288ea80466391e49192b#diff-f1f3c6677661c21128a30f64358c6615c6801043e1cbfcc904302422a17b3246R78

And AdMob API is changed to be FullScreen Event. So I removed this api.

Reply:
But I think  the change in the implementation of the  AdMob library should not be directly attach to the Plugin API.

The capacitor users still need to know that a reward was given and proxy events like "rewarded, failed", etc exist exist. 

Currently the only way is to wait for a promise call, but for angular users, events are more Rxjs Observables friendly.

This would meant a really big change for the Capacitor users.

I think we should keep those events since it is just a couple lines of code.

Reply:
> events are more Rxjs Observables friendly.

I forgot this point of view. Thanks and agree!!
But I suggest change to API name `onUserEarnedReward` .  `onAdLoaded` is a misleading name.

Reply:
I think I will enable Kotlin support so I can create clean Kotlin enums with all the events names